### PR TITLE
Fix for broken cluster shutdown 

### PR DESCRIPTION
**Pull request by johanandren**
References #30026

### CODE REVIEW

File: akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala [hunk 1]
```diff
@@ -841,8 +841,8 @@ private[cluster] class ClusterCoreDaemon(publisher: ActorRef, joinConfigCompatCh
   }
 
   def startPrepareForShutdown(): Unit = {
-    preparingForShutdown = true
     if (!preparingForShutdown) {
+      preparingForShutdown = true
```
Comment (johanandren) [inline akka-cluster/src/main/scala/akka/cluster/ClusterDaemon.scala:845]: lol, how did we miss this in review?

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
Let's merge this after validation to unlock nightlies. The repeat job was configured to run the branch, and was failing. Mail overflow I guess.

I'll can change it to this branch to confirm that we have a green run. https://jenkins.akka.io:8498/job/legacy-akka-multi-node-repeat4

**Comment by johanandren** [comment]
Repeat job is green, I'll merge.

**Pull request merged**